src/js/utils/dom.js

```diff
@@ -215,5 +216,11 @@ export function openLink(link, target, additionalOptions = {}) {
     a.href = link;
     a.target = target;
     a = Object.assign(a, additionalOptions);
-    a.dispatchEvent(new MouseEvent(`click`, { bubbles: true, cancelable: true, view: window }));
+
+    // Must support all browsers.
+    if (Browser.firefox) {
+        a.dispatchEvent(new MouseEvent('click', { bubbles: true, cancelable: true, view: window }));
+    } else {
+        a.click();
```

Comment:
Shouldn't `ie` and `.click()` be the exception and not the norm?

Reply:
Because the bug was that it was not working in Firefox. A fix was implemented (dispatchEvent), but that fix didn't work in IE. So the method was updated to use the original implementation (that worked everywhere besides Firefox), but with a fallback for Firefox.